crates/swc_ecma_minifier/src/metadata/pure_exprs.rs

```diff
@@ -0,0 +1,822 @@
+        dummy_ident_expr!(BigInt),
+        dummy_ident_expr!(DataView),
+        dummy_ident_expr!(Date),
+        dummy_ident_expr!(Error),
+        dummy_ident_expr!(EvalError),
+        dummy_ident_expr!(Event),
+        dummy_ident_expr!(EventTarget),
+        dummy_ident_expr!(Float32Array),
+        dummy_ident_expr!(Float64Array),
+        dummy_ident_expr!(Int16Array),
+        dummy_ident_expr!(Int32Array),
+        dummy_ident_expr!(Int8Array),
+        dummy_ident_expr!(Intl),
+        dummy_ident_expr!(JSON),
+        dummy_ident_expr!(Map),
+        dummy_ident_expr!(MessageChannel),
+        dummy_ident_expr!(MessageEvent),
+        dummy_ident_expr!(MessagePort),
+        dummy_ident_expr!(Promise),
+        dummy_ident_expr!(Proxy),
+        dummy_ident_expr!(RangeError),
+        dummy_ident_expr!(ReferenceError),
+        dummy_ident_expr!(Reflect),
+        dummy_ident_expr!(Set),
+        dummy_ident_expr!(Symbol),
+        dummy_ident_expr!(SyntaxError),
+        dummy_ident_expr!(TextDecoder),
+        dummy_ident_expr!(TextEncoder),
+        dummy_ident_expr!(TypeError),
+        dummy_ident_expr!(URIError),
+        dummy_ident_expr!(URL),
+        dummy_ident_expr!(URLSearchParams),
+        dummy_ident_expr!(Uint16Array),
+        dummy_ident_expr!(Uint32Array),
+        dummy_ident_expr!(Uint8Array),
+        dummy_ident_expr!(Uint8ClampedArray),
+        dummy_ident_expr!(WeakMap),
+        dummy_ident_expr!(WeakSet),
+        dummy_ident_expr!(WebAssembly),
+        dummy_ident_expr!(clearInterval),
+        dummy_ident_expr!(clearTimeout),
+        dummy_ident_expr!(console),
+        dummy_ident_expr!(decodeURI),
+        dummy_ident_expr!(decodeURIComponent),
+        dummy_ident_expr!(encodeURI),
+        dummy_ident_expr!(encodeURIComponent),
+        dummy_ident_expr!(escape),
+        dummy_ident_expr!(globalThis),
+        dummy_ident_expr!(isFinite),
+        dummy_ident_expr!(isNaN),
+        dummy_ident_expr!(parseFloat),
+        dummy_ident_expr!(parseInt),
+        dummy_ident_expr!(queueMicrotask),
+        dummy_ident_expr!(setInterval),
+        dummy_ident_expr!(setTimeout),
+        dummy_ident_expr!(unescape),
+        // Console method references are assumed to have no side effects
+        // https://developer.mozilla.org/en-US/docs/Web/API/console
+        member_expr!(DUMMY_SP, console.assert),
+        member_expr!(DUMMY_SP, console.clear),
+        member_expr!(DUMMY_SP, console.count),
+        member_expr!(DUMMY_SP, console.countReset),
+        member_expr!(DUMMY_SP, console.debug),
+        member_expr!(DUMMY_SP, console.dir),
+        member_expr!(DUMMY_SP, console.dirxml),
+        member_expr!(DUMMY_SP, console.error),
+        member_expr!(DUMMY_SP, console.group),
+        member_expr!(DUMMY_SP, console.groupCollapsed),
+        member_expr!(DUMMY_SP, console.groupEnd),
+        member_expr!(DUMMY_SP, console.info),
+        member_expr!(DUMMY_SP, console.log),
```

Comment:
I'm not sure about `console`. I often use `console.log` to debug or capture some variables to make the minifier/mangler not optimize it.

Reply:
Same here 🤣

Reply:
I used it while debugging the minifier